### ISSUE TITLE
[BUG] Unable to delete the existing folder 'ms-dotnettools.csdevkit-0.1.83-win32-x64' while installing the extension 'ms-dotnettools.csdevkit'

### ISSUE THREAD

**Issue by marleypowell** [microsoft/vscode-dotnettools#91]
Type: <b>Bug</b>

When using a VSCode profile that isn't the default profile the extension fails to install

```
2023-06-09 08:15:47.739 [error] Delete: Unable to delete the existing folder 'c:\Users\marley.powell\.vscode\extensions\ms-dotnettools.csdevkit-0.1.83-win32-x64' while installing the extension 'ms-dotnettools.csdevkit'. Please delete the folder manually and try again
    at e.ExtensionsScanner.extractUserExtension (c:\Users\marley.powell\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\code\node\sharedProcess\sharedProcessMain.js:88:12555)
    at async x.v (c:\Users\marley.powell\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\code\node\sharedProcess\sharedProcessMain.js:88:19431)
    at async x.y (c:\Users\marley.powell\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\code\node\sharedProcess\sharedProcessMain.js:88:21041)
    at async x.h (c:\Users\marley.powell\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\code\node\sharedProcess\sharedProcessMain.js:88:19006)
```

Extension version: 0.1.83
VS Code version: Code 1.78.2 (b3e4e68a0bc097f0ae7907b217c1119af9e03435, 2023-05-10T14:39:26.248Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: Yes

**Comment by alefragnani** [comment]
Hi @marleypowell ,

I have faced the exact same issue when I tried to create a new profile _after_ installing the extensions. I'm not sure this is an issue with the extension itself or the profiles feature, but I _guess_ it is an extension issue, as the profiles feature seems to handle such situations with proper message. 

As a workaround, you can create a new profile _without_ the C# extensions (none of them) and only _after_ the profile to be up and running, installs the C# extensions. Doing so, you will be able to have a new C# Profile.

Hope this helps

**Comment by timheuer** [comment]
I just tried to repro this doing:
- File > Preferences > Profiles > Create new (empty)
- navigate to extension tab and search "C#" 
- Install the "C# Dev Kit"

Success with no error.  Is it repeatable for you @marleypowell?

**Comment by AArnott** [comment]
I'm not familiar with VS Code 'profiles'. But I'm puzzled as to why _installing_ an extension would fail to _delete_ the very folder that it presumably would be installing into.

**Comment by smitpatel** [comment]
I got repro for this. Repro steps
- Install extension in default profile
- Create new profile (this won't have any extension installed)
- Install C# dev kit in this new profile.

It throws error like above because both profiles tries to install extension in same folder.

**Comment by smitpatel** [comment]
Most likely cause - We have processes running from our extension folder which are still in background after switching profile so extension cannot delete the dir.

**Comment by alefragnani** [comment]
@smitpatel, are these background processes being handled/stopped/killed on extension `deactivate`?

**Comment by marleypowell** [comment]
> I got repro for this. Repro steps
> 
> * Install extension in default profile
> * Create new profile (this won't have any extension installed)
> * Install C# dev kit in this new profile.
> 
> It throws error like above because both profiles tries to install extension in same folder.

@timheuer this is the same repro I get.

**Comment by AArnott** [comment]
I tried this repro several times. I added steps in some attempts like opening a .NET solution to ensure the extension will have been loaded. In no case did I see the error. 
I don't know why VS Code would try to uninstall the extension as part of installing that same extension to another profile, particularly if the two profiles will share an installation location and thus the new profile could have just reused the original.

I'm on VS Code 1.79.0. What version are you using, @marleypowell? Are there any other more subtle details to your repro (like loaded solutions, features used, etc? What kind of profile did you create? An empty one?
